DNN Platform/Library/Services/Installer/Installers/UrlProviderInstaller.cs

```diff
@@ -60,13 +60,8 @@ public override string AllowableFiles
         private void DeleteProvider()
         {
             try
-            {
-                //Attempt to get the Desktop Module
-                //Attempt to get the Desktop Module
-                DesktopModuleInfo desktopModule = DesktopModuleController.GetDesktopModuleByPackageID(Package.PackageID);
-
-                ExtensionUrlProviderInfo tempUrlProvider = ExtensionUrlProviderController.GetProviders(Null.NullInteger)
-                                            .SingleOrDefault(p => p.DesktopModuleId == desktopModule.DesktopModuleID);
+            {
+				ExtensionUrlProviderInfo tempUrlProvider = ExtensionUrlProviderController.GetProviders(Null.NullInteger).Where(p => p.ProviderName == _extensionUrlProvider.ProviderName && p.ProviderType == _extensionUrlProvider.ProviderType).FirstOrDefault();
```

Comment:
The installer/uninstaller reads the manifest from top to down. The natural manifest order is to list the module component before the URL extension provider component because the URL provider has a foreign key to the DesktopModule. So it makes sense that the URL extension provider component is usually listed below the module component.


Therefore, during uninstall, we cannot lookup the extension URL provider by DesktopModule (SingleOrDefault line) because by the time the uninstaller starts to the remove the extension URL provider, the DesktopModule has already been wiped off the database and it's the reason why it's currently throwing the object reference not found error. 

There is no strong argument to actually require looking up the URL extension provider by DesktopModule. The URL provider name and type should be sufficient to accurately obtain the correct provider to uninstall.


Thanks,

Stephen----------------------
www.revindex.com







---- On Tue, 16 Jun 2015 10:21:19 -0400 Brian Dukes&lt;notifications@github.com&gt; wrote ---- 

In DNN Platform/Library/Services/Installer/Installers/UrlProviderInstaller.cs:
 &gt; @@ -60,13 +60,8 @@ public override string AllowableFiles &gt; private void DeleteProvider() &gt; { &gt; try &gt; - { &gt; - //Attempt to get the Desktop Module &gt; - //Attempt to get the Desktop Module &gt; - DesktopModuleInfo desktopModule = DesktopModuleController.GetDesktopModuleByPackageID(Package.PackageID); &gt; - &gt; - ExtensionUrlProviderInfo tempUrlProvider = ExtensionUrlProviderController.GetProviders(Null.NullInteger) &gt; - .SingleOrDefault(p =&gt; p.DesktopModuleId == desktopModule.DesktopModuleID); &gt; + { &gt; + ExtensionUrlProviderInfo tempUrlProvider = ExtensionUrlProviderController.GetProviders(Null.NullInteger).Where(p =&gt; p.ProviderName == _extensionUrlProvider.ProviderName &amp;&amp; p.ProviderType == _extensionUrlProvider.ProviderType).FirstOrDefault();  Unless there's a reason otherwise, I think I'd prefer to keep this using SingleOrDefault:
 ExtensionUrlProviderInfo tempUrlProvider = ExtensionUrlProviderController.GetProviders(Null.NullInteger) .SingleOrDefault(p =&gt; p.ProviderName == _extensionUrlProvider.ProviderName &amp;&amp; p.ProviderType == _extensionUrlProvider.ProviderType);  Otherwise looks good. Do you know why Package.PackageID was throwing a null reference? Is this running after the package is uninstalled?
 —
Reply to this email directly or view it on GitHub.